lib/cn/cn_tree.c

```diff
@@ -123,6 +129,47 @@ cn_node_free(struct cn_tree_node *tn)
     }
 }
 
+static void
+cn_node_enqueue_subspill(struct subspill *ss, struct cn_tree_node *tn)
+{
+    struct list_head *p;
+    struct subspill *entry;
+
+    mutex_lock(&tn->tn_ss_lock);
+
+    /* Add ss at the right position in the node's mutation list. The list is sorted by
+     * sgen - smallest to largest.
+     */
+    list_for_each(p, &tn->tn_ss_list) {
```

Comment:
list_for_each_entry()

Reply:
See list.h, if you do not break out of the iteration the item pointer will be null at the end of the iteration.

Reply:
Right, list_for_each_entry() needs that and I've made that change. However, I don't think `list_for_each()` has that issue.

Reply:
You're right, I only changed the list_for_entry_entry* macros, list_for_each() didn't have the ubsan issue...  I can't explain the crash I saw then...